source/extensions/transport_sockets/tls/context_impl.cc

```diff
@@ -704,20 +703,20 @@ bool ContextImpl::verifySubjectAltName(X509* cert,
   return false;
 }
 
-bool ContextImpl::dnsNameMatch(const std::string& dns_name, const char* pattern) {
+bool ContextImpl::dnsNameMatch(const std::string& dns_name, const std::string_view pattern) {
   if (dns_name == pattern) {
     return true;
   }
 
-  size_t pattern_len = strlen(pattern);
+  size_t pattern_len = pattern.length();
   if (pattern_len > 1 && pattern[0] == '*' && pattern[1] == '.') {
     if (dns_name.length() > pattern_len - 1) {
       const size_t off = dns_name.length() - pattern_len + 1;
       if (Runtime::runtimeFeatureEnabled("envoy.reloadable_features.fix_wildcard_matching")) {
         return dns_name.substr(0, off).find('.') == std::string::npos &&
-               dns_name.compare(off, pattern_len - 1, pattern + 1) == 0;
+               dns_name.compare(off, pattern_len - 1, pattern.data() + 1, pattern_len - 1) == 0;
```

Comment:
I think this would be easier to read as `dns_name.substr(off, pattern_len -1) == pattern.substr(1, pattern_len - 1)`, which I think is the same thing. And it won't have an allocation if you change `dns_name` to a string_view also. But if you'd rather not, I think it is correct as-is